src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationNodes.vb

```diff
@@ -0,0 +1,1662 @@
+            _operationFactory = operationFactory
+            _value = value
+            _cases = cases
+        End Sub
+
+        Protected Overrides Function CreateValue() As IOperation
+            Return SetParentOperation(_operationFactory.Create(_value), Me)
+        End Function
+
+        Protected Overrides Function CreateCases() As ImmutableArray(Of ISwitchCaseOperation)
+            Return SetParentOperation(_operationFactory.CreateFromArray(Of BoundCaseBlock, ISwitchCaseOperation)(_cases), Me)
+        End Function
+    End Class
+
+    Friend NotInheritable Class VisualBasicLazyTryOperation
+        Inherits LazyTryOperation
+
+        Private ReadOnly _operationFactory As VisualBasicOperationFactory
+        Private ReadOnly _body As BoundNode
+        Private ReadOnly _catches As ImmutableArray(Of BoundCatchBlock)
+        Private ReadOnly _finally As BoundNode
+
+        Friend Sub New(operationFactory As VisualBasicOperationFactory, body As BoundNode, catches As ImmutableArray(Of BoundCatchBlock), [finally] As BoundNode, exitLabel As ILabelSymbol, semanticModel As SemanticModel, syntax As SyntaxNode, type As ITypeSymbol, constantValue As [Optional](Of Object), isImplicit As Boolean)
+            MyBase.New(exitLabel, semanticModel, syntax, type, constantValue, isImplicit)
+            _operationFactory = operationFactory
+            _body = body
+            _catches = catches
+            _finally = [finally]
+        End Sub
+
+        Protected Overrides Function CreateBody() As IBlockOperation
+            Return DirectCast(SetParentOperation(_operationFactory.Create(_body), Me), IBlockOperation)
+        End Function
+
+        Protected Overrides Function CreateCatches() As ImmutableArray(Of ICatchClauseOperation)
+            Return SetParentOperation(_operationFactory.CreateFromArray(Of BoundCatchBlock, ICatchClauseOperation)(_catches), Me)
+        End Function
+
+        Protected Overrides Function CreateFinally() As IBlockOperation
+            Return DirectCast(SetParentOperation(_operationFactory.Create(_finally), Me), IBlockOperation)
+        End Function
+    End Class
+
+    Friend NotInheritable Class VisualBasicLazyTupleOperation
+        Inherits LazyTupleOperation
+
+        Private ReadOnly _operationFactory As VisualBasicOperationFactory
+        Private ReadOnly _elements As ImmutableArray(Of BoundExpression)
+
+        Friend Sub New(operationFactory As VisualBasicOperationFactory, elements As ImmutableArray(Of BoundExpression), semanticModel As SemanticModel, syntax As SyntaxNode, type As ITypeSymbol, naturalType As ITypeSymbol, constantValue As [Optional](Of Object), isImplicit As Boolean)
+            MyBase.New(semanticModel, syntax, type, naturalType, constantValue, isImplicit)
+            _operationFactory = operationFactory
+            _elements = elements
+        End Sub
+
+        Protected Overrides Function CreateElements() As ImmutableArray(Of IOperation)
+            Ret
```

Comment:
>VisualBasicLazyFromEndIndexOperation [](start = 32, length = 36)

It doesn't look like this class is used. Please make sure that all added classes are actually used. We shouldn't be adding dead code.

Reply:
I did prune all of the classes that weren't being used, however.

---
In reply to: [232412331](https://github.com/dotnet/roslyn/pull/31010#discussion_r232412331) [](ancestors = 232412331,232351287)